### ISSUE TITLE
[feature] Create minor update versions for the Intel compiler in the default settings

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#9026]
The intel compiler has some behavior similar to gcc, there are minor versions, that are relevant to compatibility.

```
version: ["11", "12", "13", "14", "15", "16", "17", "18", "19", "19.1"]
```

It might be necessary or useful to represent the update too:

```
version: ["11", "12", "13", "14", "15", "16", "17", "18", "19", "19.0.3", "19.0.5"]
```

The update can be 19.0.3.203, so maybe use a ``update = [None, "ANY"]`` subsetting

@ansutremel

**Comment by memsharded** [comment]
It seems that https://github.com/conan-io/conan/issues/9304 IntelOneApi will require some changes to the Intel compiler definition. Lets handle these 2 issues together.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9522 implemented this, to be released in 1.41.
